bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -0,0 +1,536 @@
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.PING_LATENCY), newState);
+
+        newState = new QuantityType<>(Double.parseDouble(downloadBandwidth) / 125000.0, Units.MEGABIT_PER_SECOND);
+        logger.debug("downloadBandwidth: {}", newState);
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.DOWNLOAD_BANDWIDTH), newState);
+
+        newState = new QuantityType<>(Double.parseDouble(downloadBytes), Units.BYTE);
+        logger.debug("downloadBytes: {}", newState);
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.DOWNLOAD_BYTES), newState);
+
+        newState = new QuantityType<>(Double.parseDouble(downloadElapsed) / 1000.0, Units.SECOND);
+        logger.debug("downloadElapsed: {}", newState);
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.DOWNLOAD_ELAPSED), newState);
+
+        newState = new QuantityType<>(Double.parseDouble(uploadBandwidth) / 125000.0, Units.MEGABIT_PER_SECOND);
+        logger.debug("uploadBandwidth: {}", newState);
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.UPLOAD_BANDWIDTH), newState);
+
+        newState = new QuantityType<>(Double.parseDouble(uploadBytes), Units.BYTE);
+        logger.debug("uploadBytes: {}", newState);
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.UPLOAD_BYTES), newState);
+
+        newState = new QuantityType<>(Double.parseDouble(uploadElapsed) / 1000.0, Units.SECOND);
+        logger.debug("uploadElapsed: {}", newState);
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.UPLOAD_ELAPSED), newState);
+
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.INTERFACE_EXTERNALIP),
+                new StringType(String.valueOf(interfaceExternalIp)));
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.INTERFACE_INTERNALIP),
+                new StringType(String.valueOf(interfaceInternalIp)));
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.ISP),
+                new StringType(String.valueOf(isp)));
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.RESULT_URL),
+                new StringType(String.valueOf(resultUrl)));
+        updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.SERVER),
+                new StringType(String.valueOf(server)));
+    }
+
+    /*
+     * Checks to make sure the executable for speedtest is valid
+     */
+    public boolean checkConfig(String execPath) {
+        File file = new File(execPath);
+        if (!checkFileExists(file)) { // Check if entered path exists
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Speedtest executable not found. Please check configuration.");
```

Comment:
localized error messages is a nice improvement.
I'm willing to spend this effort once the binding get's closer of beeing merged.
the PR is already open since b/o 2021 :-o

Reply:
@MikeTheTux - I think we are close now. 🙂

Reply:
Are we? 🤔

Reply:
implemented. please review 🙏